pymilvus/decorators.py

```diff
@@ -35,7 +24,23 @@ def retry_on_rpc_failure(retry_times=10, wait=1, retry_on_deadline=True):
     def wrapper(func):
         @functools.wraps(func)
         def handler(self, *args, **kwargs):
+            if len(args) > 0:
+                _timeout = args[-1]
+            else:
+                _timeout = kwargs.get("timeout", None)
```

Comment:
Maybe we should check this first and then read from `args`. In `timeout` passed in `kwargs`, it won't appear in `args`.

Reply:
Good point, but whether its kwargs or args, it depends on how we call them

``` python
import functools


def wrapper(func):
    @functools.wraps(func)
    def handler(self, *args, **kwargs):
        func(self, *args, **kwargs)
        print(f"""[args]  : {args},
[kwargs]: {kwargs}\n""")
    return handler


class Test:
    def __init__(self):
        pass

    @wrapper
    def search(self, a, b=1, timeout=None, **kwargs):
        pass


if __name__ == "__main__":
    ok = Test()

    print("------------")
    print("search(0, 1)")
    ok.search(0, 1)

    print("-----------------------------")
    print("search(a=0, b=10, timeout=10)")
    ok.search(a=0, b=10, timeout=10)

    print("----------------")
    print("search(0, 1, 10)")
    ok.search(0, 1, 10)

    print("------------------------")
    print("search(0, 1, timeout=10)")
    ok.search(0, 1, timeout=10)

    print("---------------------------------------------")
    print("search(random='random', a=0, b=1, timeout=10)")
    ok.search(random="random", a=0, b=1, timeout=10)
```

```
------------
search(0, 1)
[args]  : (0, 1),
[kwargs]: {}

-----------------------------
search(a=0, b=10, timeout=10)
[args]  : (),
[kwargs]: {'a': 0, 'b': 10, 'timeout': 10}

----------------
search(0, 1, 10)
[args]  : (0, 1, 10),
[kwargs]: {}

------------------------
search(0, 1, timeout=10)
[args]  : (0, 1),
[kwargs]: {'timeout': 10}

---------------------------------------------
search(random='random', a=0, b=1, timeout=10)
[args]  : (),
[kwargs]: {'random': 'random', 'a': 0, 'b': 1, 'timeout': 10}
```

Reply:
So I'm going to change it as you suggest